docs/src/modules/ROOT/pages/using-timefold-solver/modeling-planning-problems.adoc

```diff
@@ -2840,12 +2840,162 @@ def generate_demo_data(demo_data: DemoData) -> Timetable:
 Usually, most of this data comes from your data layer,
 and your solution implementation just aggregates that data and creates the uninitialized planning entity instances for the solver to plan.
 
-[#planningModelInheritance]
-== Inheritance for Planning Entities
+[#planningModelSolutionInheritance]
+== Inheritance for Planning Solutions
 
 You can use Java's type system naturally to model your problem.
 The solver however enforces some additional restrictions
-on how inheritance can be used when it comes to planning entities and planning variables:
+on how inheritance can be used when it comes defining planning solution models:
+
+. If both the child and parent solution classes are annotated, the solver must identify and accept all the defined annotated members from each class.
+. If a child class is not annotated yet inherits from a solution class, it should be considered a solution, and the previous statement continues to apply to it.
+. If a class declares any annotated members, it must be annotated as a solution, even if a supertype already has the annotation.
+. A child solution class must not redefine any existing annotated member, but it can add new ones.
+. Inheriting annotated members from more than one parent solution class, also known as multiple inheritance, is not allowed. Therefore, only one parent solution class or one level of inheritance is allowed.
+
+=== Defining multiple annotated members
+
+According to the first rule, you can define multiple annotated members through the use of inheritance.
+The following example illustrates two planning solutions, where the child class `TestdataExtendedSolution` inherits from `TestdataBaseSolution`. Additionally, multiple annotated members, `valueList`, `entityList`, `score`, and `secondEntityList`, are accessed.
+
+[tabs]
+====
+Java::
++
+[source,java,options="nowrap"]
+----
+@PlanningSolution
+public class TestdataBaseSolution {
```

Comment:
Let's not use `Testdata` prefixes here. It is boilerplate, the code works equally well without it.

Please also remove this from the entity section, I must have missed this there during the original review.